src/main/java/com/ibm/cloudant/kafka/connect/StructToMapConverter.java

```diff
@@ -0,0 +1,94 @@
+package com.ibm.cloudant.kafka.connect;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.data.Schema.Type;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StructToMapConverter<R extends ConnectRecord<R>> {
+    
+    private static Logger LOG = LoggerFactory.getLogger(StructToMapConverter.class);
+
+    public Map<String, Object> convert(ConnectRecord<R> record) {
+        if (record.valueSchema() == null) {
+            if (record.value() instanceof Map) {
+                return (Map<String, Object>)record.value();
+            }
+        } else {
+            Map<String, Object> toReturn = new HashMap<>();
+            if (record.value() instanceof Map) {
+                return convertMap((Map)record.value(), toReturn);
+            } else if (record.value() instanceof Struct) {
+                return convertStruct((Struct)record.value(), toReturn);    
+            }
+        }
+        throw new IllegalArgumentException(String.format("Type %s not supported, are you using value.converter=org.apache.kafka.connect.json.JsonConverter?", record.value().getClass()));
```

Comment:
Adapted and used in 3fdee9d 

Note the first failure case:
```
                    throw new IllegalArgumentException(String.format("Type %s not supported with schema of type Map (or no schema)",
```
Because that leg covers both Map schema and no schema